selinux/common/typeattributes.cil

```diff
@@ -0,0 +1,106 @@
+(typeattribute common_f)
+(macro common_f ((type ARG1))
+    (roletype object_r ARG1)
+    (typeattributeset common_f ARG1)
+)
+
+(typeattribute protected_f)
+(macro protected_f ((type ARG1))
+    (roletype object_r ARG1)
+    (typeattributeset protected_f ARG1)
+)
+
+(typeattribute system_f)
+(macro system_f ((type ARG1))
+    (roletype object_r ARG1)
+    (typeattributeset system_f ARG1)
+)
+
+(typeattribute system_socket_f)
+(macro system_socket_f ((type ARG1))
+    (roletype object_r ARG1)
+    (typeattributeset system_socket_f ARG1)
+)
+
+(allow system_socket_f tmpfs_t (filesystem (associate)))
+
+(typeattribute common_device_f)
+(macro common_device_f ((type ARG1))
+    (roletype object_r ARG1)
+    (typeattributeset common_device_f ARG1)
+)
+
+(typeattribute protected_device_f)
+(macro protected_device_f ((type ARG1))
+    (roletype object_r ARG1)
+    (typeattributeset protected_device_f ARG1)
+)
+
+(typeattribute device_f)
+(typeattributeset device_f device_t)
+(typeattributeset device_f common_device_f)
+(typeattributeset device_f protected_device_f)
+
+(typeattribute any_f)
+(typeattributeset any_f filesystem_f)
+(typeattributeset any_f common_f)
+(typeattributeset any_f protected_f)
+(typeattributeset any_f system_f)
+(typeattributeset any_f system_socket_f)
+(typeattributeset any_f device_f)
+
+(typeattribute unconfined_f)
+(typeattributeset unconfined_f tmpfs_t)
+(typeattributeset unconfined_f common_f)
+(typeattributeset unconfined_f common_device_f)
+
+(typeattribute filesystem_f)
+(macro filesystem_f ((type ARG1))
+    (roletype object_r ARG1)
+    (typeattributeset filesystem_f ARG1)
+)
+
+(allow filesystem_f self (filesystem (associate)))
+(allow filesystem_f fs_t (filesystem (associate)))
+
+(typeattribute system_service_p)
+(macro system_service_p ((type ARG1))
+    (roletype system_r ARG1)
+    (typeattributeset system_service_p ARG1)
+)
+
+(typeattribute system_container_p)
+(macro system_container_p ((type ARG1))
+    (roletype system_r ARG1)
+    (typeattributeset system_container_p ARG1)
+)
+
+(typeattribute client_p)
+(macro client_p ((type ARG1))
+    (roletype client_r ARG1)
+    (typeattributeset client_p ARG1)
+)
+
+(typeattribute pod_p)
+(macro pod_p ((type ARG1))
+    (roletype client_r ARG1)
+    (typeattributeset pod_p ARG1)
+)
+
+(typeattribute any_p)
+(typeattributeset any_p kernel_t)
+(typeattributeset any_p init_t)
+(typeattributeset any_p system_service_p)
+(typeattributeset any_p system_container_p)
+(typeattributeset any_p client_p)
+(typeattributeset any_p pod_p)
+
+(typeattribute system_p)
+(typeattributeset system_p kernel_t)
+(typeattributeset system_p init_t)
+(typeattributeset system_p system_service_p)
+(typeattributeset system_p client_p)
```

Comment:
Some stuff like `client_p` doesn't make much sense to be separate from `system_service_p`, so perhaps the number of typeattributes could be reduced to ease understanding, visualization and possibly reduce policy size and overhead